src/wallet/wallet.cpp

```diff
@@ -853,6 +853,67 @@ bool CWallet::LoadUnifiedAddressMetadata(const ZcashdUnifiedAddressMetadata &add
                 addrmeta.GetReceiverTypes());
 }
 
+PaymentAddress CWallet::GetPaymentAddressForRecipient(const uint256& txid, const libzcash::RecipientAddress recipient) const {
+    auto recipientsPtr = sendRecipients.find(txid);
+
+    // TODO: We should scrub change addresses from the results here.
```

Comment:
Open an issue for this, blocking at a minimum 5.0.0, and ideally 4.7.0. Change addresses must not ever be shown to users, per our previous discussions about UX; if we ever show a change address to a user, we have to assume it can receive external funds, which breaks a bunch of the UX properties we're after. We can _maybe_ get away without anyone noticing or starting to rely on this in 4.7.0, but if it is the behaviour of 5.0.0 then we're screwed from a UX perspective.

What we need to do is replace any change address in the UX with a corresponding external address (a sender UA involved in the transaction that we can identify as coming from the same UFVK, or the default account address as a fallback).

Reply:
This has now been fixed.